src/main/java/com/impactupgrade/nucleus/service/segment/StripeDataSyncService.java

```diff
@@ -42,7 +54,51 @@ public void syncContacts(Calendar updatedAfter) throws Exception {
   public void syncTransactions(Calendar updatedAfter) throws Exception {
     List<CrmDonation> crmDonations = env.primaryCrmService().getDonations(updatedAfter);
     for (CrmDonation crmDonation : crmDonations) {
-      //TODO: bulk update in Stripe
+      //TODO: one by one update in Stripe
     }
   }
+
+  protected Customer updateOrCreateCustomer(CrmContact crmContact) throws Exception {
+    Optional<Customer> existingCustomer = getCustomer(crmContact);
+    if (existingCustomer.isPresent()) {
+      // Update
+      CustomerUpdateParams customerUpdateParams = CustomerUpdateParams.builder()
+          .setName(crmContact.getFullName())
+          .setEmail(crmContact.email.toLowerCase(Locale.ROOT))
+          .setAddress(toUpdateAddress(crmContact.mailingAddress))
+          .build();
+      return env.stripeClient().updateCustomer(existingCustomer.get(), customerUpdateParams);
+    } else {
+      // Create
+      CustomerCreateParams.Builder customerCreateParamsBuilder = CustomerCreateParams.builder()
+          .setName(crmContact.getFullName())
+          .setEmail(crmContact.email.toLowerCase(Locale.ROOT))
+          .setAddress(toCreateAddress(crmContact.mailingAddress));
+      return env.stripeClient().createCustomer(customerCreateParamsBuilder);
```

Comment:
Oh interesting! Hrm, comment this out for now? I like the idea, but the Stripe sync already creates customers as needed, and there's more logic to it. But we should combine some of these concepts soon...